### PR TITLE
[Merged by Bors] - feat(CategoryTheory/Limits): Add basic pullback lemmas

### PR DESCRIPTION
**Pull request by callesonne**
This PR adds some basic lemmas about pullbacks of a monos/epi by itself. This is used in #14208.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by erdOne** [comment]
Thanks!
maintainer merge

File: Mathlib/CategoryTheory/CommSq.lean [hunk 1]
```diff
@@ -109,6 +109,14 @@ lemma vert_comp {W X Y Y' Z Z' : C} {f : W ⟶ X} {g : W ⟶ Y} {g' : Y ⟶ Y'}
     CommSq f (g ≫ g') (h ≫ h') i' :=
   flip (horiz_comp (flip hsq₁) (flip hsq₂))
 
+theorem mono {W X Y : C} {f : W ⟶ X} {g : W ⟶ X} {i : X ⟶ Y} [Mono i] (sq : CommSq f g i i) :
```
Comment (joelriou) [inline Mathlib/CategoryTheory/CommSq.lean:112]: I would prefer a name like `cancel_mono` or `eq_of_mono`.

Comment (callesonne) [reply Mathlib/CategoryTheory/CommSq.lean:112]: great suggestion, fixed!

File: Mathlib/CategoryTheory/Limits/Shapes/Pullback/CommSq.lean [hunk 2]
```diff
@@ -362,6 +362,18 @@ lemma of_iso (h : IsPullback fst snd f g)
               rw [← reassoc_of% commfst, e₂.hom_inv_id, Category.comp_id]
             · change snd = e₁.hom ≫ snd' ≫ e₃.inv
               rw [← reassoc_of% commsnd, e₃.hom_inv_id, Category.comp_id]))⟩
+section
+
+variable {P X Y : C} {fst : P ⟶ X} {snd : P ⟶ X} {f : X ⟶ Y} [Mono f]
+
+lemma fst_iso_of_mono_eq (h : IsPullback fst snd f f) : IsIso fst :=
```
Comment (joelriou) [inline Mathlib/CategoryTheory/Limits/Shapes/Pullback/CommSq.lean:369]:
```suggestion
lemma isIso_fst_of_mono (h : IsPullback fst snd f f) : IsIso fst :=
```

Comment (joelriou) [reply Mathlib/CategoryTheory/Limits/Shapes/Pullback/CommSq.lean:369]: Same for `snd_iso_of_mono_eq`, `inl/rnr_iso_of_epi_eq`.

File: Mathlib/CategoryTheory/Limits/Shapes/Pullback/Mono.lean [hunk 3]
```diff
@@ -191,30 +191,43 @@ section
...
+theorem pullbackSymmetry_hom_of_mono_eq : (pullbackSymmetry f f).hom = 𝟙 _ := by
   ext
   · simp [fst_eq_snd_of_mono_eq]
   · simp [fst_eq_snd_of_mono_eq]
 
-instance fst_iso_of_mono_eq [Mono f] : IsIso (pullback.fst f f) := by
-  refine ⟨⟨pullback.lift (𝟙 _) (𝟙 _) (by simp), ?_, by simp⟩⟩
-  ext
+variable {f} in
+lemma PullbackCone.fst_iso_of_mono_eq {t : PullbackCone f f} (ht : IsLimit t) :
```
Comment (joelriou) [inline Mathlib/CategoryTheory/Limits/Shapes/Pullback/Mono.lean:213]: Similar changes below:
```suggestion
lemma PullbackCone.isIso_fst_of_mono_of_isLimit {t : PullbackCone f f} (ht : IsLimit t) :
```
Names of instances should also contain `isIso`.

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**